### PR TITLE
Share position with external recipient

### PR DESCRIPTION
**Pull request by mjoyce91**
Adds functionality from #120 :
- Integrates with new email API endpoint to send a position by email (note: real emails are not actually sent yet, pending email server)
- Runs a regular expression to check that email ends with "state.gov" and alerts the user if their input does not end as such
- Confirmation that email was sent is alerted to user

### CODE REVIEW

File: src/Components/Share/Share.test.jsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import thunk from 'redux-thunk';
+import Share from './Share';
+
+const middlewares = [thunk];
+const mockStore = configureStore(middlewares);
+
+describe('Main', () => {
+  const api = 'http://localhost:8000/api/v1';
+
+  beforeEach(() => {
```
Comment (burgwyn) [inline src/Components/Share/Share.test.jsx]: Do we need this?

File: src/Components/Share/ShareButton/ShareButton.jsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+      recipient: '',
+      warning: false,
+      timeout: false,
+    };
+  }
+
+  changeEmail(e) {
+    const recipient = e.target.value;
+    this.setState({ recipient });
+    const warning = !(/@state.gov\s*$/.test(recipient));
```
Comment (burgwyn) [inline src/Components/Share/ShareButton/ShareButton.jsx]: We should externalize this

Comment (burgwyn) [inline src/Components/Share/ShareButton/ShareButton.jsx]: And confirm the pattern

File: src/reducers/share.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+export function shareHasErrored(state = false, action) {
+  switch (action.type) {
+    case 'SHARE_HAS_ERRORED':
```
Comment (burgwyn) [inline src/reducers/share.js]: I like the consistency of this pattern.  Do you think it is worth tracking some additional events, like _IS_STARTING?

File: src/Components/Share/Share.test.jsx (hunk 1 above)
Comment (mjoyce91) [inline src/Components/Share/Share.test.jsx]: Removed

File: src/Components/Share/ShareButton/ShareButton.jsx (hunk 2 above)
Comment (mjoyce91) [inline src/Components/Share/ShareButton/ShareButton.jsx]: Good point, there may be sub domains or alternate domains for valid State employee email addresses.

**Pull request merged**